sklearn/decomposition/online_lda.py

```diff
@@ -0,0 +1,653 @@
+
+        if cal_diff:
+            # This step finishes computing the sufficient statistics for the M step
+            component_diff = np.zeros(self.components_.shape)
+            for comp_diff in comp_diffs:
+                component_diff += comp_diff
+            component_diff *= self.exp_dirichlet_component_
+        else:
+            component_diff = None
+
+        return (doc_topic_distr, component_diff)
+
+    def _em_step(self, X, total_samples, batch_update):
+        """
+        EM update for 1 iteration
+        update `_component` by bath VB or online VB
+
+        parameters
+        ----------
+        X : sparse matrix, shape = [n_samples, n_features]
+            Document word matrix.
+
+        total_samples : integer
+            Total umber of document. It is only used when
+            batch_update is `False`.
+
+        batch_update : boolean
+            parameter that control updating method.
+            `True` for batch learning, `False` for online learning
+
+        Returns
+        -------
+        doc_topic_distr : array, shape = [n_samples, n_topics]
+            Unnormailzed document topic distribution.
+        """
+
+        # E-step
+        doc_topic_distr, component_diff = self._e_step(X, cal_diff=True)
+
+        # M-step
+        if batch_update:
+            self.components_ = self.topic_word_prior + component_diff
+        else:
+            # online update
+            # In literature, the weight is `rho`
+            weight = np.power(self.learning_offset + self.n_iter_, -self.learning_decay)
+            doc_ratio = float(total_samples) / X.shape[0]
+            self.components_ *= (1 - weight)
+            self.components_ += (weight *
+                                 (self.topic_word_prior + doc_ratio * component_diff))
+
+        # update `component_` related variables 
+        self.dirichlet_component_ = _dirichlet_expectation(self.components_)
+        self.exp_dirichlet_component_ = np.exp(self.dirichlet_component_)
+        self.n_iter_ += 1
+        return doc_topic_distr
+
+    def _to_csr(self, X, whom):
+        """
+        check & convert X to csr format, and make sure no negative value in X.
+
+        parameters
+        ----------
+        X :  array-like
+        
+        """
+        X = check_array(X, accept_sparse='csr')
+        check_non_negative(X, whom)
+        if not sp.issparse(X):
+            X = sp.csr_matrix(X)
+
+        return X
+
+    def fit_transform(self, X, y=None):
+        """
+        Learn a model for X and returns the transformed data
+
+        Parameters
+        ----------
+        X : array or sparse matrix, shape = [n_samples, n_features]
+            Document word matrix.
+
+        Returns
+        -------
+        doc_topic_distr : array, [n_samples, n_topics]
+            Topic distribution for each document.
+        """
+        X = self._to_csr(X, "LatentDirichletAllocation.fit_transform")
+        return self.fit(X).transform(X)
```

Comment:
this could be implemented more efficiently by skipping the additional E-step, right?
Or is that instantaneous anyhow?

You could rename ``fit`` to ``_fit``, return ``doc_topics`` in ``_fit``, here just return ``_fit`` and in ``fit`` just call ``_fit`` and then ``return self``.